### PR TITLE
Add Copy File Path button to the context menu of file list

### LINKED ISSUE #2944: View/copy full path of file (Changes or History tab)
### Description

In history mode, view/copy complete file path if it's longer than the given screenspace. View screenshot. Copying gives me `src/main/webapp/WEB-INF/views/web/p…/manage.jsp`

See #2801 for more context.

**GitHub Desktop version:** 0.9.1

**OS version:** 10.12.6

**Actual behavior:**

copy-file

**Reproduces how often:** 100%

### PR DESCRIPTION
**Pull request by Amabel**
This adds a 'Copy File Path' button in the file list context menu in both Changes and History tabs. When click the button, it will copy the fullpath of the selected file to your clipboard.

- [x] add copy button in Changes tab

- [x] add copy button in History tab

Fixes #2944

Here are the GIFs which show the behavior:

|Changes tab|History tab|
|---|---|
|||

### CODE REVIEW

File: app/src/ui/lib/context-menu.ts [hunk 1]
```diff
@@ -1,4 +1,6 @@
 const RestrictedFileExtensions = ['.cmd', '.exe', '.bat', '.sh']
+export const CopyFilePathLabel = 'Copy File Path'
```
Comment (niik) [inline app/src/ui/lib/context-menu.ts]: Could we please conditionalize this for Windows? macOS expects Title Case but for Windows it should be Sentence case.

```suggestion
export const CopyFilePathLabel = __DARWIN__ ? 'Copy File Path' : 'Copy file path' 
```

File: app/src/ui/history/file-list.tsx [hunk 2]
```diff
@@ -132,6 +134,11 @@ export class FileList extends React.Component<IFileListProps, {}> {
       : DefaultEditorLabel
 
     const items = [
+      {
+        label: CopyFilePathLabel,
+        action: () => clipboard.writeText(fullPath),
+        enabled: fileExistsOnDisk,
```
Comment (niik) [inline app/src/ui/history/file-list.tsx]: Copying the path should probably be allowed even if the file doesn't exist on disk, could be useful for people trying to look up the file on the web for example or for those who want to use it on the command line to restore it or check out a previous version.

File: app/src/ui/changes/changes-list.tsx [hunk 3]
```diff
@@ -345,6 +347,14 @@ export class ChangesList extends React.Component<
 
     items.push(
       { type: 'separator' },
+      {
+        label: CopyFilePathLabel,
+        action: () => {
+          const fullPath = Path.join(this.props.repository.path, path)
+          clipboard.writeText(fullPath)
+        },
+        enabled: status !== AppFileStatus.Deleted,
```
Comment (niik) [inline app/src/ui/changes/changes-list.tsx]: Copying the path should probably be allowed even if the file doesn't exist on disk, could be useful for people trying to look up the file on the web for example or for those who want to use it on the command line to restore it or check out a previous version.

**Comment by Amabel** [comment]
@niik Thanks for the review and comments. I've fixed this according to your suggestions!

**Pull request merged**